src/go/configgenerator/route_generator.go

```diff
@@ -330,13 +351,40 @@ func makeRoute(routeMatcher *routepb.RouteMatch, method *configinfo.MethodInfo,
 			},
 		},
 		Decorator: &routepb.Decorator{
+			// TODO(taoxuy@): check if the generated span name length less than the limit.
 			// Note we don't add ApiName to reduce the length of the span name.
 			Operation: fmt.Sprintf("%s %s", util.SpanNamePrefix, method.ShortName),
 		},
 	}
 }
 
-func makeCatchAllUnmatchedRoute() *routepb.Route {
+func makeMethodNotAllowedRoute(methodNotAllowedRouteMatcher *routepb.RouteMatch, uriTemplateInSc string) *routepb.Route {
+	spanName := fmt.Sprintf("%s UnknownHttpMethodForPath_%s", util.SpanNamePrefix, uriTemplateInSc)
+
+	if len(spanName) > util.SpanNameMaxByteNum {
+		newSpanName := fmt.Sprintf("%s UnknownHttpMethod", util.SpanNamePrefix)
```

Comment:
How about just truncate the string to MaxByte and end it with "..." to indicate it has been truncated.

Reply:
Done